meerk40t/core/elements.py

```diff
@@ -5540,6 +5541,79 @@ def contains(box, x, y=None):
         self._emphasized_bounds = None
         self.set_emphasis(None)
 
+    def classify_legacy(self, elements, operations=None, add_op_function=None):
+        """
+        Classify does the placement of elements within operations.
+        "Image" is the default for images.
+        Typically,
+        If element strokes are red they get classed as cut operations
+        If they are otherwise they get classed as engrave.
+        However, this differs based on the ops in question.
+        :param elements: list of elements to classify.
+        :param operations: operations list to classify into.
+        :param add_op_function: function to add a new operation, because of a lack of classification options.
+        :return:
+        """
+        if elements is None:
+            return
+        reverse = self.context.classify_reverse
+        if reverse:
+            elements = reversed(elements)
+        if operations is None:
+            operations = list(self.ops())
+        if add_op_function is None:
+            add_op_function = self.add_op
+        for element in elements:
+            was_classified = False
+            if hasattr(element, "operation"):
+                add_op_function(element)
+                continue
+            if element is None:
+                continue
+            for op in operations:
+                if op.operation == "Raster":
+                    if element.stroke is not None and op.color == abs(element.stroke):
+                        op.add(element, type="opnode")
+                        was_classified = True
+                    elif isinstance(element, SVGImage):
```

Comment:
This `elif` section should be removed. It is a bug that results in images in Raster ops.

Reply:
That only happens if there is no image op. Which seems like the best choice in that case. You might want the image.

Reply:
Nope. Put the raster operation first and watch it duplicate. Create a disabled Image op if need be.